boa/src/builtins/regexp/mod.rs

```diff
@@ -391,12 +364,12 @@ impl RegExp {
         if let Some(object) = this.as_object() {
             if let Some(regexp) = object.borrow().as_regexp() {
                 return Ok(JsValue::new(match flag {
-                    'g' => regexp.global,
-                    'm' => regexp.multiline,
-                    's' => regexp.dot_all,
-                    'i' => regexp.ignore_case,
-                    'u' => regexp.unicode,
-                    'y' => regexp.sticky,
+                    'g' => regexp.flags.contains(RegExpFlags::GLOBAL),
```

Comment:
Could we match bytes instead of characters here? Should be more efficient

Reply:
In my last commit the code matches on bytes, but this could be replaced with the `RegExpFlags` which would allow for the following reduction in matching code:

```rust
fn regexp_has_flag(this: &JsValue, flag: RegExpFlags, context: &mut Context) -> JsResult<JsValue> {
    if let Some(object) = this.as_object() {
        if let Some(regexp) = object.borrow().as_regexp() {
            return Ok(JsValue::new(regexp.flags.contains(flag)));
        }
        // ...
    }
    // ...
} 
```

Which would be more readable?